packages/optimizely-cms-sdk/src/react/server.tsx

```diff
@@ -328,13 +338,142 @@ export function getPreviewUtils(opti: OptimizelyComponentProps['opti']) {
+          url = input.item?.Url ?? input.url?.default ?? '';
+        }
+      }
+
+      return appendPreviewToken(url);
+    },
+
+    /**
+     * Generates a responsive image srcset from ContentReference renditions.
+     * Automatically deduplicates renditions with the same width and appends preview tokens.
+     *
+     * @param input - ContentReference from a DAM asset with renditions
+     * @returns Comma-separated srcset string with width descriptors (e.g., "url1 100w, url2 500w")
+     *
+     * @example
+     * ```tsx
+     * const { src, srcset } = getPreviewUtils(opti);
+     *
+     * <img
+     *   src={src(opti.image)}
+     *   srcSet={srcset(opti.image)}
+     *   sizes="(max-width: 768px) 100vw, 50vw"
+     *   alt="Responsive image"
+     * />
+     * ```
+     */
+    srcset(input: InferredContentReference | null | undefined): string {
+      if (!input?.item || !('Renditions' in input.item)) return '';
+
+      const renditions = input.item.Renditions;
+      if (!renditions || renditions.length === 0) return '';
+
+      // Track seen widths to avoid duplicate width descriptors
+      const seenWidths = new Set<number>();
+
+      const srcsetEntries = renditions
+        .filter((r) => {
+          if (!r.Url || !r.Width) return false;
+          // Skip if we've already seen this width
+          if (seenWidths.has(r.Width)) return false;
+          seenWidths.add(r.Width);
+          return true;
+        })
+        .map((r) => {
+          const url = appendPreviewToken(r.Url!);
+          return `${url} ${r.Width}w`;
+        });
+
+      return srcsetEntries.join(', ');
+    },
+
+    /**
+     * Extracts the alt text from a ContentReference DAM asset.
+     * Returns fallback text if AltText is empty, null, or not available.
+     *
+     * @param input - ContentReference from a DAM asset
+     * @param fallback - Optional fallback text to use when AltText is not available
+     * @returns The alt text or fallback, or empty string if neither is available
+     *
+     * @example
+     * ```tsx
+     * const { alt } = getPreviewUtils(opti);
+     *
+     * // Use AltText from asset
+     * <img src={src(opti.image)} alt={alt(opti.image)} />
+     *
+     * // With fallback
+     * <img src={src(opti.image)} alt={alt(opti.image, 'Default description')} />
+     *
+     * // Use Title as fallback
+     * <img src={src(opti.image)} alt={alt(opti.image, opti.image.item?.Title ?? '')} />
+     * ```
+     */
+    alt(
+      input: InferredContentReference | null | undefined,
+      fallback?: string
+    ): string {
+      if (!input) return fallback ?? '';
+
+      // Check if item has AltText property (PublicImageAsset or PublicVideoAsset)
+      if (input.item && 'AltText' in input.item) {
+        const altText = input.item.AltText ?? '';
+        return altText || (fallback ?? '');
       }
-      return url;
+
+      return fallback ?? '';
```

Comment:
I think this `alt` function should not belong here (`getPreviewUtils`) because it has nothing to do with preview